lib/node_modules/@stdlib/_tools/eslint/rules/tsdoc-doctest/README.md

```diff
@@ -0,0 +1,184 @@
+
+## Usage
+
+```javascript
+var rule = require( '@stdlib/_tools/eslint/rules/tsdoc-doctest' );
+```
+
+#### rule
+
+[ESLint rule][eslint-rules] to ensure that return annotations in TSDoc examples match the actual output. The rule validates `@example` blocks in TSDoc comments within `.d.ts` files.
+
+**Bad**:
+
+<!-- eslint-disable stdlib/tsdoc-doctest -->
+
+```typescript
+/**
+* Adds two numbers.
+*
+* @param x - first number
+* @param y - second number
+* @returns sum of x and y
+*
+* @example
+* var result = add( 2, 3 );
+* // returns 6
+*/
+declare function add( x: number, y: number ): number;
+
+export = add;
+```
+
+**Good**:
+
+```typescript
+/**
+* Adds two numbers.
+*
+* @param x - first number
+* @param y - second number
+* @returns sum of x and y
+*
+* @example
+* var result = add( 2, 3 );
+* // returns 5
+*/
+declare function add( x: number, y: number ): number;
+
+export = add;
+```
+
+</section>
+
+<!-- /.usage -->
+
+<section class="examples">
+
+## Examples
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var Linter = require( 'eslint' ).Linter;
+var parser = require( '@typescript-eslint/parser' );
+var rule = require( '@stdlib/_tools/eslint/rules/tsdoc-doctest' );
+
+var linter = new Linter();
+
+// Register the TypeScript parser and ESLint rule:
+linter.defineParser( '@typescript-eslint/parser', parser );
+linter.defineRule( 'tsdoc-doctest', rule );
+
+// Generate our source code with incorrect return annotation:
+var code = [
+    '/**',
+    '* Returns the absolute value of a number.',
+    '*',
+    '* @param x - input value',
+    '* @returns absolute value',
+    '*',
+    '* @example',
+    '* var result = abs( -3 );',
+    '* // returns 2',
+    '*/',
+    'declare function abs( x: number ): number;',
+    '',
+    'export = abs;'
+].join( '\n' );
+
+// Lint the code:
+var result = linter.verify( code, {
+    'parser': '@typescript-eslint/parser',
+    'parserOptions': {
+        'ecmaVersion': 2018,
+        'sourceType': 'module'
+    },
+    'rules': {
+        'tsdoc-doctest': 'error'
+    }
+}, {
+    'filename': 'lib/node_modules/@stdlib/math/base/special/abs/docs/types/index.d.ts'
+});
+/* returns
+    [
+        {
+            'ruleId': 'tsdoc-doctest',
+            'severity': 2,
+            'message': 'Displayed return value is `2`, but expected `3` instead',
+            'line': 9,
+            'column': 1,
+            'nodeType': null,
+            'endLine': 10,
+            'endColumn': 37
+        }
+    ]
+*/
+```
+
+</section>
+
+<!-- /.examples -->
+
+<section class="notes">
+
+## Notes
+
+-   The corresponding JavaScript package must be loadable via `require('@stdlib/<package>')`
+-   The rule skips validation if the package cannot be loaded.
+-   Examples are executed in a sandboxed VM context with limited globals.
+-   The rule validates assignment-form examples (e.g., `var x = fn(...);` followed by an annotation). It does not validate console output or expression-only forms using `// =>`.
```

Comment:
Why this limitation?

Reply:
Added support for linting annotations of form `// =>`.